### PR TITLE
GraphQL: Mutations for managing wish lists (EE)

### LINKED ISSUE #7866: GraphQL: Add schema coverage for multiple wishlists
# Community engineering issue

Add documentation for the following mutations

createWishlist
deleteWishlist
copyProductsToWishlist
moveProductsToWishlist
updateWishlist

The `customer` query will need to be updated as well.

Type of Doc:

-  [x] New topic
-  [ ] Topic Update, Devdoc link:
-  [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/28551

Magento 2 GitHub PR: https://github.com/magento/partners-magento2ee/pull/248

Internal ticket: MC-37628

Magento 2 version:

-  [x] 2.4.2

Notifications of Changes:

-  [ ] Changes to Admin interface?
-  [ ] New software requirements?
-  [x] New REST endpoints or updated GraphQL query syntax?
-  [ ] New command line option?

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) adds documentation for the following mutations:

* createWishlist
* updateWishlist
* deleteWishlist
* moveProductsToWishlist
* copyProductsToWishlist

~~It does not update the storeConfig attributes because adding this information now would cause a massive merge conflict with #7853~~

Fixes #7866 

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

whatsnew
Added mutations to manage multiple wish lists on Magento Commerce installations.

### CODE REVIEW

File: src/guides/v2.4/graphql/mutations/copy-products-to-wishlist.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+`wishlistItems` | [WishlistItemCopyInput!]! | An array containing a set of wish list IDs that are to be copied
+
+### WishlistItemCopyInput attributes {#WishlistItemCopyInput}
+
+The WishlistItemCopyInput object contains the following attributes.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`quantity` | Float | The quantity of this item to move to the destination wish list. This value cannot be greater than the quantity in the source wish list
+`wishlist_item_id` | ID! | The ID of the item to be moved
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/copy-products-to-wishlist.md:126]: This one will be copied 🙂

```suggestion
`wishlist_item_id` | ID! | The ID of the item to be copied
```

File: src/guides/v2.4/graphql/mutations/copy-products-to-wishlist.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+## Output attributes
+
+The `UpdateProductsInWishlistOutput` object can contain the following attributes.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`user_errors` | [[WishListUserInputError!](#WishListUserInputError)] | An array of errors encountered while adding products to a wish list
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/copy-products-to-wishlist.md:134]: In the current implementation the resolver will return empty array if there are no errors 

```suggestion
`user_errors` | [[WishListUserInputError!](#WishListUserInputError)]! | An array of errors encountered while adding products to a wish list
```

File: src/guides/v2.4/graphql/mutations/move-products-to-wishlist.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+`wishlistId` | ID! | The ID of the wish list to update
+`wishlistItems` | [WishlistItemMoveInput!]! | An array containing a set of wish list IDs that are to be moved
+
+### WishlistItemMoveInput attributes {#WishlistItemMoveInput}
+
+The WishlistItemMoveInput object contains the following attributes.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`quantity` | Float | The quantity of this item to move to the destination wish list. This value cannot be greater than the quantity in the source wish list
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/move-products-to-wishlist.md:97]: If the quantity is not set, the entire row item will be moved to the new wish list. 
Having this info in the documentation might be very helpful. 

What do you think?

**Review by BarnyShergold — APPROVED** [review]
I've carefully checked this for English syntax & grammar (US) and all look perfect! Amazing work :)

File: src/guides/v2.4/graphql/mutations/move-products-to-wishlist.md (hunk 3 above)
Comment (keharper) [reply src/guides/v2.4/graphql/mutations/move-products-to-wishlist.md:99]: I moved such a statement to the top. I don't want the text here to vary too much from the `description` strings.

**Review by dobooth — APPROVED** [review]

**Review by prabhuram93 — APPROVED** [review]
Looks Good! @keharper

**Comment by keharper** [comment]
running tests

**Comment by jeff-matthews** [comment]
@keharper, should there be links in the whatsnew?

**Pull request merged**